crates/aranya-quic-channels/src/lib.rs

```diff
@@ -0,0 +1,293 @@
+#![warn(missing_docs)]
+
+//! An implementation of the syncer using QUIC.
+
+use std::{net::SocketAddr, sync::Arc};
+
+use anyhow::{anyhow, Result};
+use aranya_fast_channels::Label;
+use bytes::Bytes;
+use heapless::Vec as HVec;
+use s2n_quic::{
+    client::Connect,
+    connection,
+    provider::{self, StartError},
+    stream::{self, BidirectionalStream, ReceiveStream, SendStream},
+    Client, Connection, Server,
+};
+use tokio::{
+    select,
+    sync::{
+        mpsc::{self, error::TryRecvError},
+        Mutex as TMutex,
+    },
+};
+use tracing::{debug, error};
+
+/// An error running the AQC client
+#[derive(Debug, thiserror::Error)]
+pub enum AqcError {
+    /// A channel was closed.
+    #[error("channel closed")]
+    ChannelClosed,
+    /// An error creating a quic connection.
+    #[error("connect error: {0}")]
+    Connect(#[from] connection::Error),
+    /// An error using a stream.
+    #[error("stream error: {0}")]
+    Stream(#[from] stream::Error),
+    /// An error starting an s2n quic client.
+    #[error("start error: {0}")]
+    Start(#[from] StartError),
+    /// An infallible error.
+    #[error("infallible error: {0}")]
+    Infallible(#[from] std::convert::Infallible),
+    /// An io error.
+    #[error("io error: {0}")]
+    Io(#[from] std::io::Error),
+    /// An internal AQC error.
+    #[error("internal error: {0}")]
+    Internal(#[from] anyhow::Error),
+}
+
+/// Runs a server listening for quic channel requests from other peers.
+pub async fn run_channels(client: Arc<TMutex<AqcClient>>, mut server: Server) {
+    loop {
+        select! {
+            Some(conn) = server.accept() => {
+                match handle_connection(conn).await {
+                    Ok(Some(channel)) => {
+                        if client
+                            .lock()
+                            .await
+                            .new_channels
+                            .push(channel).is_err() {
+                            error!("Channel full. Unable to insert channel");
```

Comment:
Include the error returned from `push()` in the trace:
```suggestion
error!("Unable to insert channel: {:?}", e);
```